### PR TITLE
CADC-12927 fix node permission inconsistencies

### PR DESCRIPTION
**Pull request by jburke-cadc**

### CODE REVIEW

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/GetNodeAction.java [hunk 1]
```diff
@@ -147,65 +147,68 @@ public void doAction() throws Exception {
         
         if (serverNode instanceof ContainerNode) {
             ContainerNode node = (ContainerNode) serverNode;
-
             log.debug("node: " + node);
 
-            // TBD: sorting parameters
```
Comment (pdowler) [inline cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/GetNodeAction.java:153]: Why all these deletes? This diff is impossible to grok 

If I read this correctly, the rejection of unsupported operations are indented some more so not always checked... rejection should always happen as soon as possible and not conditionally.

File: cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java [hunk 2]
```diff
@@ -95,15 +107,20 @@ public class TransferTest extends org.opencadc.conformance.vos.TransferTest {
...
     static {
         Log4jInit.setLevel("org.opencadc.conformance.vos", Level.INFO);
-        
         Log4jInit.setLevel("org.opencadc.vospace", Level.INFO);
         Log4jInit.setLevel("org.opencadc.cavern", Level.INFO);
     }
     
     public TransferTest() {
         super(Constants.RESOURCE_ID, Constants.TEST_CERT);
+
```
Comment (pdowler) [inline cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java:116]: check cavern NodesTest setup and move things to Constants for consistency

**Review by pdowler — CHANGES_REQUESTED** [review]

File: cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/GetNodeAction.java (hunk 1 above)
Comment (jburke-cadc) [reply cadc-vos-server/src/main/java/org/opencadc/vospace/server/actions/GetNodeAction.java:153]: The code actually changed was minimal. Didn't think to look at the diff. Yes in indentation makes it hard to read. A couple of blocks of code were moved as well which also doesn't help.

The 'if' for the indentation is checking if the caller has permission to list the children, and if they do, do the following that gives the node a iterator over its child nodes. If they don't have permission, then skip getting the child iterator. After that, write the node.

File: cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java (hunk 2 above)
Comment (jburke-cadc) [reply cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java:116]: Updated.

File: cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java [hunk 3]
```diff
@@ -173,4 +183,108 @@ public void testPreauthToken() {
...
+     * Test that a user can update a data node they do not own, but have write permission to the node.
+     *
+     * The authSubject user is the owner of the parent ContainerNode and child DataNode.
+     * The altSubject user is a member of the altGroup group.
+     * The altGroup by default does not have write access to the parent or child nodes.
+     *
+     * The altGroup group is given write access to the child node, which should
+     * allow the altSubject to write to the child node.
+     */
+    @Test
```
Comment (pdowler) [inline cavern/src/intTest/java/org/opencadc/cavern/TransferTest.java:197]: ok, so this method is here but the "enable" is in the test suite? I assume the rework is incomplete...

**Review by pdowler — CHANGES_REQUESTED** [review]

**Pull request closed without merging**